graphql_compiler/schema_transformation/utils.py

```diff
@@ -263,3 +284,84 @@ def check_ast_schema_is_valid(ast):
 
     query_type = get_query_type_name(schema)
     visit(ast, CheckQueryTypeFieldsNameMatchVisitor(query_type))
+
+
+def is_property_field(selection):
+    """Return True if selection is a property field, False if a vertex field or type coercion.
+
+    Args:
+        selection: type Field or InlineFragment. An element occuring inside a SelectionSet
+
+    Returns:
+        True if the selection is a property field, False if it's a vertex field or type coercion
+    """
+    if isinstance(selection, ast_types.InlineFragment):
+        return False
+    if isinstance(selection, ast_types.Field):
+        if (
+            selection.selection_set is None or
+            selection.selection_set.selections is None or
+            selection.selection_set.selections == []
+        ):
+            return True
+        else:
+            return False
+    else:
+        raise AssertionError(
+            u'Input selection "{}" is not of type Field or InlineFragment.'.format(selection)
+        )
+
+
+class CheckQueryIsValidToSplitVisitor(Visitor):
+    """Check the query only has supported directives, and its fields are correctly ordered."""
+    # This is very restrictive for now. Other cases (e.g. tags not crossing boundaries) are
+    # also ok, but temporarily not allowed
+    supported_directives = frozenset(('filter', 'output', 'optional'))
+
+    def enter_Directive(self, node, *args):
```

Comment:
nits: capitalization, let's make this `enter_directive` and likewise for other functions.

Reply:
resolved offline

~why would python let you do this~